ui/components/app/import-token/token-list/token-list.component.js

```diff
@@ -14,60 +30,104 @@ export default class TokenList extends Component {
   };
 
   render() {
     const {
       results = [],
       selectedTokens = {},
+
       onToggleToken,
       tokens = [],
+      currentNetwork,
+      testNetworkBackgroundColor,
     } = this.props;
 
     return results.length === 0 ? (
-      <TokenListPlaceholder />
+      <Box paddingLeft={4} paddinfRight={4} className="emptyListTokens">
+        <TokenListPlaceholder />
+      </Box>
     ) : (
-      <div className="token-list">
-        <div className="token-list__title">
-          {this.context.t('searchResults')}
-        </div>
-        <div className="token-list__tokens-container">
-          {Array(6)
+      <Box className="token-list">
+        <Box className="token-list__tokens-container">
+          {Array(12)
             .fill(undefined)
             .map((_, i) => {
               const { symbol, name, address } = results[i] || {};
               const tokenAlreadyAdded = checkExistingAddresses(address, tokens);
+              let tokenAlreadySelected;
+              if (tokenAlreadyAdded) {
+                tokenAlreadySelected = results[i];
+              }
               const onClick = () =>
                 !tokenAlreadyAdded && onToggleToken(results[i]);
-
               return (
                 Boolean(results[i]?.iconUrl || symbol || name) && (
-                  <div
-                    className={classnames('token-list__token', {
-                      'token-list__token--selected': selectedTokens[address],
-                      'token-list__token--disabled': tokenAlreadyAdded,
+                  <Box
+                    key={address}
+                    display={Display.Flex}
+                    className={classnames('token-list__token_component', {
+                      'token-list__token_component--selected':
+                        selectedTokens[address],
+                      'token-list__token_component--disabled':
+                        tokenAlreadyAdded,
                     })}
                     onClick={onClick}
-                    onKeyPress={(event) => event.key === 'Enter' && onClick()}
-                    key={i}
-                    tabIndex="0"
                   >
-                    <div
-                      className="token-list__token-icon"
-                      style={{
-                        backgroundImage:
-                          results[i]?.iconUrl && `url(${results[i]?.iconUrl})`,
-                      }}
-                    />
-                    <div className="token-list__token-data">
-                      <span className="token-list__token-name">{`${name} (${symbol})`}</span>
-                    </div>
-                  </div>
+                    <Box display={Display.Flex} alignItems={AlignItems.center}>
+                      <Checkbox
+                        isChecked={
+                          selectedTokens[address] ||
+                          tokenAlreadySelected ||
+                          false
```

Comment:
I think if we get to `|| false` we'd have a falsy value already, so I think `|| false` can be removed`